rpcs3/Input/ds4_pad_handler.cpp

```diff
@@ -235,26 +222,26 @@ std::unordered_map<u64, u16> ds4_pad_handler::get_button_values(const std::share
 	if (!ds4_dev)
 		return keyBuffer;
 
-	auto buf = ds4_dev->padData;
+	const ds4_input_report_common& input = *reinterpret_cast<const ds4_input_report_common*>(ds4_dev->padData.data());
```

Comment:
This is undefined behavior, use std::memcpy, read_from_ptr, write_to_ptr and std::bit_cast only when dealing with these.

Reply:
Such as 

```
ds4_input_report_common temp{};
temp.. do stuff;
write_to_ptr<ds4_input_report_common>(ds4_dev->padData.data(), &temp);

Reply:
I dont't really see the point in doing that for 1 byte aligned structs.
But I rewrote it to use a union for the input reports anyway.

Reply:
union is still ub though

Reply:
I don't see how anything in the current code is undefined

Reply:
Its ub, accessing more than one object through the same memory location. In c++, union is not about seamless casting, it provides lifetime for the first object written to it.
Its called "strict type aliasing rule"

Reply:
Should be fixed now